### DISCUSSION TITLE
How does client generation work for production?

### DISCUSSION THREAD

**Discussion by BelugaBeanie** [prisma/prisma#13263] [Q&A]
Hi all, first off I just wanna say prisma is awesome! Regarding client generation, the official docs say prisma should be a dev dependency, how does that work when I come to deploy to production.

Surely I need the CLI to generate the client. Or am I supposed to bundle the client somehow with the dist? Any input on this topic would be amazing as I couldn't find much online without having to install all dev dependencies just to deploy.

Thanks in advance.

**Comment by austincrim** [comment]
Hey @BelugaBeanie 👋, we are glad that you are enjoying Prisma so far!

While the `@prisma/client` package should be a production dependency, you are correct that the `prisma` CLI package should be a development dependency. Most users run `prisma generate` in their CI pipeline or as a `prebuild` step before deploying their app. Once the client is generated, you do not need the `prisma` CLI package during runtime in production.

What does your production deployment workflow look like?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
